### DISCUSSION TITLE
Read state of capslock Keyboard with Arduino

### DISCUSSION THREAD

**Discussion by enomic-bot1** [earlephilhower/arduino-pico#2061] [Q&A]
Hello,
i would build an Keyboard, the problem is i can´t read the states of the Capslock, Numlock an scrolllock. i need this to set my leds.
I use the Keyboard.h lib, but i cant find a function in this Library, for my Problem.
Thanks for help.

**Comment by earlephilhower** [comment]
There is an `onLED` callback you can add which will let your app keep track of the states yourself.  See
https://github.com/earlephilhower/Keyboard/blob/c0f474695f83d63209fe4ad73ec5d90f27cd9e64/src/HID_Keyboard.h#L201

**Reply by enomic-bot1** [reply]
Thanks for the fast answer.
Can you write an example for this function. I dont check this.

**Reply by earlephilhower** [reply]
https://github.com/earlephilhower/arduino-pico/blob/master/libraries/Keyboard/examples/KeyboardPassword/KeyboardPassword.ino

**Comment by enomic-bot1** [comment]
Thank you, it work.
